src/persist-client/src/inspect.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! CLI introspection tools for persist
+
+use crate::r#impl::state::ProtoStateRollup;
+use crate::{Metrics, ShardId};
+use mz_ore::metrics::MetricsRegistry;
+use mz_persist::cfg::ConsensusConfig;
+use mz_persist::location::SeqNo;
+use prost::Message;
+use serde_json::json;
+
+/// Prints the current state of a given shard as JSON
+pub async fn print_current_state(
+    shard_id: ShardId,
+    consensus_uri: &str,
+) -> Result<(), anyhow::Error> {
```

Comment:
I have a preference that this be more something like `fetch_current_state` and return a `impl serde::Serialize`. then the logic for how it gets printed (serde_json + stdout) can live in the binary code and then the deps (serde_json) can be a dev-dependency

Reply:
ooo that's so sneaky, I love it